src/s3/util/fileViewerManager.ts

```diff
@@ -181,8 +223,24 @@ export class S3FileViewerManager {
     }
 
     public createTargetPath(fileNode: S3FileNode): Promise<string> {
```

Comment:
Could you document some expected inputs and outputs from this? It's a little tricky to read thanks to the slice statement.